src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -74,6 +76,20 @@
             utilityFunctions)
             .flatMap(Set::stream).collect(Collectors.toSet());
 
+    private HashMap<String, Integer> generatedIds = new HashMap<>();
```

Comment:
Use interface `Map` instead?

Reply:
This is private field, so knowing detailed type could be beneficial in common case, but here Map will work as well, changed.